test/parallel/test-stream-buffer-list.js

```diff
@@ -15,16 +15,21 @@ assert.strictEqual(emptyList.join(','), '');
 
 assert.deepStrictEqual(emptyList.concat(0), Buffer.alloc(0));
 
+const buf = Buffer.from('foo');
+
 // Test buffer list with one element.
 const list = new BufferList();
-list.push('foo');
+list.push(buf);
+
+const copy = list.concat(3);
 
-assert.strictEqual(list.concat(1), 'foo');
```

Comment:
@mcollina this is wrong whether or not the change in BufferList.js is applied.

The `n` argument of `BufferList.prototype.concat()` is not the number of `Buffer` instances in the list, but their total length when concatenated.